src/test/java/org/folio/rest/impl/PasswordResourceTest.java

```diff
@@ -13,102 +13,326 @@
+  private static final String STATE = "state";
+  private static final String VALIDATION_PATH_QUERYABLE = "/password/validators?type=";
+  private static final String VALIDATOR_RULE_TYPE_REGEXP = "RegExp";
+
   private static final String TENANT = "diku";
-  private static final String VALIDATOR_RULE_TYPE_STRONG = "strong";
-  private Vertx vertx;
-  private int port;
+  private static final String VALIDATION_RULES_TABLE_NAME = "validation_rules";
 
-  @Before
-  public void setUp(final TestContext context) {
+  private static Vertx vertx;
+  private static int port;
+
+
+  @Rule
+  public Timeout rule = Timeout.seconds(180);  // 3 minutes for loading embedded postgres
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) throws IOException {
+    Async async = context.async();
     vertx = Vertx.vertx();
     port = NetworkUtils.nextFreePort();
+
+    PostgresClient.setIsEmbedded(true);
+    PostgresClient.getInstance(vertx).startEmbeddedPostgres();
+    TenantClient tenantClient = new TenantClient("localhost", port, TENANT, TENANT);
+
     final DeploymentOptions options = new DeploymentOptions().setConfig(new JsonObject().put(HTTP_PORT, port));
-    vertx.deployVerticle(RestVerticle.class.getName(), options, context.asyncAssertSuccess());
+    vertx.deployVerticle(RestVerticle.class.getName(), options, res -> {
+      try {
+        tenantClient.post(null, res2 -> {
+          async.complete();
+        });
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    });
+  }
+
+  @AfterClass
+  public static void tearDownClass(final TestContext context) {
+    Async async = context.async();
+    vertx.close(context.asyncAssertSuccess(res -> {
+      PostgresClient.stopEmbeddedPostgres();
+      async.complete();
+    }));
   }
 
   @After
-  public void tearDown(final TestContext context) {
-    vertx.close(context.asyncAssertSuccess());
+  public void tearDown(TestContext context) throws Exception {
+    PostgresClient.getInstance(vertx, TENANT).delete(VALIDATION_RULES_TABLE_NAME, new Criterion(), event -> {
+      if (event.failed()) {
+        context.fail(event.cause());
+      }
+    });
   }
 
   @Test
-  public void shouldReturnValidatorsWithStrongRuleType(final TestContext context) {
-    //TODO Replace testing stub
-    final Async async = context.async();
-    final String url = HOST + port;
-    final String validatorsUrl = url + VALIDATION_PATH_QUERYABLE + VALIDATOR_RULE_TYPE_STRONG;
-
-    final Handler<HttpClientResponse> handler = response -> {
-      context.assertEquals(response.statusCode(), HttpStatus.SC_OK);
-      context.assertEquals(response.headers().get(HEADER_CONTENT_TYPE), APPLICATION_JSON);
-      response.bodyHandler(body -> {
-        context.assertTrue(body.toJsonObject().equals(getRuleCollectionStub()));
-        async.complete();
+  public void shouldReturnBadRequestStatusWhenPasswordIsAbsentInBody(TestContext context) {
```

Comment:
plz add javadoc for each test with description

Reply:
I don't think it is necessary as names of test methods are self-explaining.